daemon/network.go

```diff
@@ -560,38 +560,39 @@ func (daemon *Daemon) deleteNetwork(nw *libnetwork.Network, dynamic bool) error
 }
 
 // GetNetworks returns a list of all networks
-func (daemon *Daemon) GetNetworks(filter filters.Args, config types.NetworkListConfig) ([]types.NetworkResource, error) {
-	networks := daemon.getAllNetworks()
-
-	list := make([]types.NetworkResource, 0, len(networks))
+func (daemon *Daemon) GetNetworks(filter filters.Args, config types.NetworkListConfig) (networks []types.NetworkResource, err error) {
 	var idx map[string]*libnetwork.Network
 	if config.Detailed {
 		idx = make(map[string]*libnetwork.Network)
 	}
 
-	for _, n := range networks {
+	allNetworks := daemon.getAllNetworks()
+	networks = make([]types.NetworkResource, 0, len(allNetworks))
+	for _, n := range allNetworks {
 		nr := buildNetworkResource(n)
-		list = append(list, nr)
+		networks = append(networks, nr)
 		if config.Detailed {
 			idx[nr.ID] = n
 		}
 	}
 
-	var err error
-	list, err = internalnetwork.FilterNetworks(list, filter)
+	networks, err = internalnetwork.FilterNetworks(networks, filter)
 	if err != nil {
 		return nil, err
 	}
 
 	if config.Detailed {
-		for i := range list {
-			np := &list[i]
-			buildDetailedNetworkResources(np, idx[np.ID], config.Verbose)
-			list[i] = *np
+		for i := range networks {
+			np := &networks[i]
+			np.Containers = buildContainerAttachments(idx[np.ID])
+			if config.Verbose {
+				np.Services = buildServiceAttachments(idx[np.ID])
+			}
+			networks[i] = *np
```

Comment:
I'll have a look; not sure if we can pass the network around as value, because there's mutexes in it, but the local variable _probably_ is not needed no. I _think_ this may have been an attempt at updating the network atomically, but as it's a pointer anyway, there's nothing atomic here.

Reply:
Okay; updated this. Couldn't pas `nw` to `buildContainerAttachments`, because `networks` is the `types.NetworkResource` (so the "API response for networks") and not the actual _network_. But it's a bit cleaner at least;

```go
for i, nw := range networks {
	networks[i].Containers = buildContainerAttachments(idx[nw.ID])
	if config.Verbose {
		networks[i].Services = buildServiceAttachments(idx[nw.ID])
	}
}
```

It would probably be cleaner if `networks` was a `[]*types.NetworkResource` (pointer) that way we could just update it without doing the `networks[i]`, but that requires changing the `GetNetworks()` signature, so I kept that for a later exercise.